### ISSUE TITLE
bossfetch targets from distinct datasets

### ISSUE THREAD

**Issue by dmargala** [dkirkby/bossdata#6]
My target list contains entries from v5_7_0 and v5_7_2 (see http://www.sdss.org/dr12/data_access/bulk/#OpticalSpectraVersions for an explanation of these two datasets). The bossfetch program did not complain about any failed download attempts for the v5_7_2 targets. The saved files that contain html with an error 404 message.

An example v5_7_2 target is 7517-56772-0952 (plate-mjd-fiber).

I'm not sure what the right approach to handling mixed datasets is but the 404 thing seems like an issue.

**Comment by dkirkby** [comment]
I can reproduce this problem with:

```
printf "PLATE MJD FIBER\n7517 56772 0952\n" > missing.dat
bin/bossfetch missing.dat
```

which does not complain that the requested plate does not exist but saves a FITS file that is actually an HTML 404 page:

```
head -1 /Data/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/7517/spec-7517-56772-0952.fits
<!DOCTYPE html>
```

**Comment by dkirkby** [comment]
I updated `remote.download` to check the request status code and raise a RuntimeError. Now this fails with:

```
% rm /Data/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/7517/spec-7517-56772-0952.fits
% bin/bossfetch missing.dat
Skipping file with download error: /sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/7517/spec-7517-56772-0952.fits.
WARNING: 1 of 1 observations were not fetched. Re-run the command after any problems are fixed.
```

**Comment by dkirkby** [comment]
Fixed in 0.2
